crates/edr_solidity_tests/src/multi_runner.rs

```diff
@@ -349,16 +371,43 @@ impl<
                 );
 
                 for (_, arena) in &mut result.traces {
-                    let mut trace_identifier =
-                        TraceIdentifiers::new().with_local(&self.known_contracts);
                     decoder.identify(arena, &mut trace_identifier);
                     tokio::task::block_in_place(|| {
                         handle.block_on(decode_trace_arena(arena, &decoder));
                     });
                 }
+
+                if let Some(gas_report) = &mut gas_report {
```

Comment:
```suggestion
                if let Some(gas_report) = gas_report.as_mut() {
```

This should allow you to avoid the `mut: Option` in the function signature. `.as_mut()` is the way to get mutable access to the internals, whereas `&mut gas_report` would give you a mutable reference to the `Option`, hence why the function signature requires `mut`

Reply:
IIUC `.as_mut()` still requires the variable to be mutable - converts `&mut Option<>` to `Option<&mut>`, so I can't get rid of the `mut` in signature.

In any case, `as_mut()` seems better suited here, since it explicitly gives access to the internals.